### DISCUSSION TITLE
Problem with the execution with reqnroll

### DISCUSSION THREAD

**Discussion by esteban2050** [reqnroll/Reqnroll#308] [Q&A]
I am facing one problem when I am trying to execute my test written with reqnroll, I am having in this moment 2 features with 2 respective stepDefinition classes. I understand for stepDefinition classes, I need to put [Binding] at the beginning  but I have in a clase called Base one method with [BeforeScenario] to start the browser. The problem is when I am re using some steps from clase2 in clase1, the [Binding] is causing that [beforeScenario] executes multiple times and my browser is tarting many times. Could you help me with this ? I will appretiate it

**Comment by clrudolphi** [comment]
You will likely want to investigate use of a [BeforeFeature] or [BeforeTestRun] hook rather than the [BeforeScenario] hook.
See the [documentation](https://docs.reqnroll.net/latest/automation/hooks.html) for Hooks.

**Reply by esteban2050** [reply]
I understan the BeforeScenario is to execute something that I need before each scenario. For that reason I put that tag in startBrowser method that I have to initialize the browser... and BeforeFeature is to execute something one time per feature. In this case If I put beforeFuture, this method will initialize the browser only once and the other scenario will execute themselve in the same instance of the browser and I dont need that

**Reply by gasparnagy** [reply]
@esteban2050 I recommend looking at our sample app: https://github.com/reqnroll/Sample-ReqOverflow/tree/main/ReqOverflow.Specs.WebUI
